### DISCUSSION TITLE
Connect to dev server

### DISCUSSION THREAD

**Discussion by Tweekism** [jannis-baum/Vivify#126] [Q&A] [answered]
Hey Peeps,

Another day another dumb question... So I was previously doing this fine, I'm not sure what has happened.

I run `yarn dev` then `yarn viv README.md` it says its going to connect on port 3000 but them it doesn't, the browser pops up connected to 31622.

What incredibly obviously thing am i doing wrong today? Lol.

![image](https://github.com/user-attachments/assets/3ba30f80-224f-46a8-afc1-de1f5a05d037)

**Comment by jannis-baum** [comment]
Strange! I assume you are on the latest commit on `main` and have run `yarn`?

I can't reproduce this but one thing that strikes out from the screenshots is that your development server has actually stopped for whatever reason (Note "`clean exit - waiting for changes before restarting`" from `nodemon`). But even then running `yarn viv README.md` should still start a development server on port 3000 and open the file there. So not sure what's going on to be honest.

I would suggest investigating why the server might have stopped (maybe you killed it trying to kill the non-development server?) and if that doesn't lead anywhere printing the `address` in `src/app.ts` and checking where it points.

**Comment by Tweekism** [comment, marked as answer]
Hey @jannis-baum 

Ok peeps are calling on me to go play warzone with them lol. However I think I found the issue and I *think* it's a bug. **Remember I'm completely new to javascript / typescript, so go easy on me lol 😅**

Is there a way on GitHub to reference lines of code in a comment?

Anyway in `parser/config.ts` on line 60 there is an if statement first added in commit 07296ea (tag: bug) feat(#90): improve conf retrieval. Which is:-

    if (config === undefined) return defaultConfig;

Meaning if you don't have a config at `~/.vivify.json` or `~/.vivify/config.json` (mine is currently renamed to `.vivify.json.off`) then the function returns and never gets to the section that overrides with the environment variables starting on line 73

```
    // environment overrides
    for (const [env, key] of envConfigs) {
        if (process.env[env]) config[key] = process.env[env];
    }
    return config;
```

Are you able to replicate if by renaming / moving your vivify config temporarily and then running the yarn dev thing?

**Reply by jannis-baum** [reply]
Ah, yes, good spot! That's it. I will fix it ^^ Thanks a lot for finding that!

**Reply by jannis-baum** [reply]
Fixed in #128!

**Reply by Tweekism** [reply]
I may not know how to code, but I can cook up a mad git bisect...

**Reply by jannis-baum** [reply]
Hahaha🔥

Since this was a potentially quite annoying bug I made a new minor release: https://github.com/jannis-baum/Vivify/releases/tag/v0.2.1

**Comment by tuurep** [comment]
Wanna see something funny? I'm trying to debug why some neovim operators are misbehaving

So this slipped in my configs:

![image](https://github.com/user-attachments/assets/e5f78b44-6a67-4e20-80e8-40f7a422ea1b)

Bagbiwum, what a name :smile:

**Reply by Tweekism** [reply]
How did that get in there?

**Reply by tuurep** [reply]
Using init.lua as a test buffer for https://github.com/echasnovski/mini.operators
